### ISSUE TITLE
Create a way to show more details to an event

### ISSUE THREAD

**Issue by Yolgie** [boudicca-events/boudicca.events#298]
Currently we have much more information about events than we can show.

One way of making this additional info visible could be to on selecting an event or clicking on a details button to make this tile bigger and add addional fields that we want to display.

Coordinate with the UUSC Fellow for UX on what to show and how to do this.

**Comment by mahdikhashan** [comment]
Wireframe for the UI design link: https://excalidraw.com/#json=zIGQ_UzEHFelbd0hsIZKT,ZkABIJCI4WR2_IYjnH2yYQ

for the 1st iteration, I have only used `even.description` and `event.tags`.


![Image](https://github.com/boudicca-events/boudicca.events/assets/58775404/b642a581-c81b-452e-81ae-3e022b6e17d7)

**Comment by kadhonn** [comment]
it looks quite nice but i also want something which really shows ALL the properties of the event, especially those not handled otherwise.

also, maybe we could split up the a11y icon into a nicer list somehow?

**Comment by mahdikhashan** [comment]
would you share the properties we want to show here? also, I agree with you about splitting up the a11y icon, I'll find a way to show it. 

also, I need to figure out the appropriate layout for the component.

**Comment by kadhonn** [comment]
by default we should at least show all properties that we can filter for

and then somewhere else also all properties we don't show anywhere ese

**Comment by mahdikhashan** [comment]
ok, then let me first implement the first iteration and then add the properties we are filtering for. to see how layouting can be improved.

**Comment by mahdikhashan** [comment]
after researching about the effect of this feature on the user experience, I have reached to a point to continue by implementing it either as a dialog or modal to prevent layout shift. 

It also can affect a11y, since some user may find it unpleasant to follow the shift in layout.

so, the decision is to instead of injecting the details in the events row, when the user clicks or taps an event element open a modal and show the properties. 

@kadhonn do you have any input?

references: 

1. https://web.dev/articles/cls
2. https://developer.chrome.com/docs/crux/methodology/metrics#cls-metric

**Comment by mahdikhashan** [comment]
Second iteration with modal, and properties from filter drawer



![Image](https://github.com/boudicca-events/boudicca.events/assets/58775404/490c6cb8-069e-412a-ac23-990ac900ab2e)

**Comment by mahdikhashan** [comment]
Third Iteration with event drawer



![Image](https://github.com/boudicca-events/boudicca.events/assets/58775404/39b10c92-4d36-4931-9749-f7bc6f3610fe)

**Comment by kadhonn** [comment]
well, the two links you posted about cls are interesting but are only talking about unexpected layout shifts
clicking and expanding an event is not unexpected, is it?

**Comment by mahdikhashan** [comment]
it talks about expected shifts under "User-initiated layout shifts" section of the first link. but, every layout shift has an effect on the UX and A11y, expected or unexpected. 

let me make sure it does not affect A11Y first, then UX second. and I'm guessing that a drawer or modal is more predictable than an expandable element in this scenario for a person who requires accessibility. since we are already using a drawer for filters, it may make more sense to stick to the current behaviour and not introduce new one (regarding the [predictability](https://www.w3.org/WAI/fundamentals/accessibility-principles/#predictable) of the navigation in a11y). 

it can also be a valid case to be discussed with the integriert studieren institute.

I can also implement it and we can check it later with UX fellow against the personas.

**Comment by mahdikhashan** [comment]
the expanding feature would be like this, did we mean to cover the whole row with an expanded event?

https://github.com/boudicca-events/boudicca.events/assets/58775404/37f36600-82b8-4686-9f02-de6c9689bbe6

**Comment by mahdikhashan** [comment]
we talked in our daily today, our concerns at the moment are: white spaces, details that we want to embed in the element and a11y.

**Comment by mahdikhashan** [comment]
in our meeting today we decided to keep the portotypes in separate branches so then we can work on them at the same time.

**Comment by mahdikhashan** [comment]
one a11y concerns for the Modal and Drawer prototypes is to prevent mouse or keyboard trap.

**Comment by mahdikhashan** [comment]
what am I doing in the code that I'm not happy with is keeping the propeties with data attributes, I think it is not clean and there could be a better way.

also, I can not append the eventHandler to the events that are loaded after the I press the "mehr laden" button.

**Comment by kadhonn** [comment]
> we may need to consider extending the PictureProxy to have a functionality to request pictures in requested size.

that should not be hard, and also we can do that easly in another step as enhancement

**Comment by kadhonn** [comment]
> what am I doing in the code that I'm not happy with is keeping the propeties with data attributes, I think it is not clean and there could be a better way.

i wrote in your PR already, it is better to prerender the modal/drawer content for each event but keep it invisible

> also, I can not append the eventHandler to the events that are loaded after the I press the "mehr laden" button.

why not?
it should work to simple retrigger the registering after the load more code is done

**Comment by mahdikhashan** [comment]
You are right, I'll prerender modal/drawer content. I'll implement the retriggering functionality.

**Comment by mahdikhashan** [comment]
some progress with modal

https://github.com/boudicca-events/boudicca.events/assets/58775404/78121c03-e03e-43c8-9283-b3b36e1b2c9f

**Comment by mahdikhashan** [comment]
https://github.com/boudicca-events/boudicca.events/assets/58775404/b84d5b60-ac17-4ad8-b9aa-6912c05686f2

**Comment by mahdikhashan** [comment]
event details drawer

https://github.com/boudicca-events/boudicca.events/assets/58775404/c2a02ce1-ab37-4920-9946-edc3898bd4f7

**Comment by mahdikhashan** [comment]
two-column layout:



![Image](https://github.com/boudicca-events/boudicca.events/assets/58775404/151ee49a-0394-478e-ac68-af845f6eb31c)

**Comment by mahdikhashan** [comment]
in the drawer, I prefer to have a two column layout instead of the single column I used  for the modal.

**Comment by twatzl** [comment]
Personally I prefer the two column layout. Possibly because that is what I used for museumsbahn-events.at as well. 😅

I think a modal always introduces a new layer in the UI that is not really needed. The drawer also has this kind of overlaying effect. With the two column layout the information stays "on the same level". I hope this makes sense.

btw maybe a thing that should also be considered is what should happen when you get sent a link to a specific event from someone. Like:
https://museumsbahn-events.at/events/2024-04-30T22_00_00.000Z-01-mai-2024-saisoner%C3%B6ffnung-im-eisenbahn--und-bergbaumuseum---lokpark-ampflwang

**Comment by mahdikhashan** [comment]
the latest update about this issue:

1. we talked about asking feedback from outside people, I'll wait till friday for a response from my guy.
2. I would also see if I can ask any other fellows for any feedback or help.

@kadhonn if I missed anything here, please let me know.

**Comment by p4dd9** [comment]
* the preview image as a clickable is not clear what happens
* keep an eye on accessible modals/drawers if done so (https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/) - restore tab where the user came from (e.g. event focus from item 3, should be restored after closing dialog/drawer)
* re-using filter-drawer: i would not re-se a reserved ui context which focus should be filtering and searching to make the ui clear what part is used for what
* i would also challenge our previous decision of not having directlinks/detailpages: imo, singleview by now, make sense - as user-flow, i might want to share details about the boudicca data of an event that could happen via
--> additional data in the url query on the searchpage
--> have a singleview!
--> imo the focus should be that boudicca is the entry point to go to an external link with the purpose to help a user if the event is accessible before opening an external page


TL;DR i think a singleview would be nice (altought SPA logic will break if a new doc loads when navigating) with focus on external open and other recomemndations for events on the bottom

**Comment by kadhonn** [comment]
i think this is a good point about having a single-view: we can still make sure this single-view is accessible and has information about accessibility and people can share this single-view. this information can be lost when we only allow sharing the original url

but i am not sure if a single view is the best solution while browsing our result list. if every time i want to see more information about this event i need to open a new tab this could get annoying.

i still think in regards to usability the expanding event is the best option, because i can have multiple events open in parallel and it does not disrupt my scrolling. i think the main issue we had is that it creates gaps in our grid, maybe we can find simple solutions for that?
like getting rid of the grid, however this would look like then?
or filling it with a placeholder?
or accepting the gaps?

**Comment by mahdikhashan** [comment]
latest update on the issue: 

1. the code need refactoring with respect to change requests on the open PR.
2. the modal now has a keyboard trap
3. tabbable table height should be fixed.
